lib/executor/constant_arrival_rate.go

```diff
@@ -249,6 +252,30 @@ func (car ConstantArrivalRate) Run(ctx context.Context, out chan<- stats.SampleC
 		return activeVU
 	}
 
+	remainingUnplannedVUs := maxVUs - preAllocatedVUs
+	makeUnplannedVUCh := make(chan struct{})
+	defer close(makeUnplannedVUCh)
+	go func() {
+		defer close(returnedVUs)
+		defer func() {
+			// this is done here as to not have an unplannedVU in the middle of initialization when
+			// starting to return activeVUs
+			for i := uint64(0); i < atomic.LoadUint64(&activeVUsCount); i++ {
+				<-activeVUs
+			}
+		}()
+		for range makeUnplannedVUCh {
+			car.logger.Debug("Starting initialization of an unplanned VU...")
+			initVU, err := car.executionState.GetUnplannedVU(maxDurationCtx, car.logger)
+			if err != nil {
+				// TODO figure out how to return it to the Run goroutine
+				car.logger.WithError(err).Error("Error while allocating unplanned VU")
+			}
+			car.logger.Debug("The unplanned VU finished initializing successfully!")
```

Comment:
This message would be confusing if "Error while allocating unplanned VU" is logged before it, and worse still: wouldn't we be passing a `nil` `initVU` to `activateVU()` in that case? :thinking:

Maybe add a `continue` in the `err != nil` block and let's address this `TODO` now?

Reply:
I propose that a test that fails with this code is added as well

Reply:
Resolved by https://github.com/loadimpact/k6/pull/1500/commits/07dea1284be2c9a58c8d06f8399dd4519bc50b61?

I don't want to add test for something so specific that isn't handled properly at this time and we know we'll change. We should fix the TODO above when we also fix the "not spawn a goroutine for every iteration" issue, and fix the "share code between arrival-rate executors" issue...

Reply:
OK, LGTM. The `err != nil` branch is not covered according to Codecov, but I'm fine with adding the test later if this will change.